model/model.go

```diff
@@ -221,6 +223,34 @@ func (t *TableInfo) ColumnIsInIndex(c *ColumnInfo) bool {
 	return false
 }
 
+// PartitionType is the type for PartitionInfo
+type PartitionType int
+
+// Partition types.
+const (
+	PartitionTypeRange PartitionType = 1
+	PartitionTypeHash  PartitionType = 2
+	PartitionTypeList  PartitionType = 3
+)
+
+// PartitionInfo provides table partition info.
+type PartitionInfo struct {
+	Type    PartitionType
+	Expr    string
+	Columns []CIStr
+
+	Definitions []PartitionDefinition
+}
+
+// PartitionDefinition defines a single partition.
+type PartitionDefinition struct {
+	ID       int64
+	Name     string
+	LessThan []string
```

Comment:
`LessThan` is only used in range partition ?

Reply:
Yes

Reply:
so how do we define a hash or list or key partition ?

Reply:
It would be implemented later... @zz-jason

Reply:
I think we should implement a more extensible struct to define partition info.

Reply:
It's compatible when we add new fields later.